edgexfoundry/app-service-configurable
# refactor!: Replace internal topics from config with new constants and use base topic

BREAKING CHANGE: Internal topics no longer configurable, except the base topic. Trigger topics for edgex-messagebus and external-mqtt now directly under Trigger section. All configured topics (subscribe and function pipeline) now automatically have the base topic (default of "edgex/") prepended.







**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/app-service-configurable/blob/main/.github/CONTRIBUTING.md

## PR Checklist
Please check if your PR fulfills the following requirements:

  TBD

## Testing Instructions
Run non-secure edgex stack with Device Virtual and MQTT Broker 
build ASC from this branch with `-o -cp` options
Run ASC with each profile (skip push-to-core and functional-tests)
Verify from logs that data is flowing from Device Virtual for the following profiles
- sample
   - Also verify `float-pipeline` and `int8-16-pipeline` execute when the appropriate events are received
- rules-engine
   - app-rules-engine in docker must be stopped 
   - events received are published back to `edgex/rules-events` topic
- mqtt-export
  -  verify data is export to the MQTT Broker
- http-export
  -  ignore export errors since export URL doesn't have a listener
For the verify logs for following profiles:
- metrics-influxdb
   - Bootstraps and waiting for events on   `edgex/telemetry/#`
- external-mqtt-trigger
  - Bootstraps, connects to MQTT Broker 
  - Logs contains `Subscribed to topic(s) 'external-request/#' for MQTT trigger`
  - 
## New Dependency Instructions (If applicable)
